tls/src/main/java/org/bouncycastle/tls/TlsProtocol.java

```diff
@@ -153,6 +154,8 @@ protected boolean isTLSv13ConnectionState()
     private volatile boolean resumableHandshake = false;
     private volatile int appDataSplitMode = ADS_MODE_1_Nsub1;
 
+    private int soTimeout = -1;
```

Comment:
Ist der Defaultwert -1 so glücklich? _Eigentlich_ werden negative Werte für Timeouts an andere Stellen verboten, sieh z.B.  `java.net.Socket#setSoTimeout`. 

In dieser Klasse ist nur der explizite Wert `0` eine Bedeutung, insofern _könnte_ das so passen.